src/viam/examples/motor/example_motor.cpp

```diff
@@ -1,67 +1,127 @@
+            robot = vs::RobotClient::at_address(robot_address, options);
+            cout << "Successfully connected to the robot" << endl;
+        } catch (const std::exception& e) {
+            cerr << "Failed to connect to the robot. Exiting." << endl;
+            throw;
+        }
 
-    std::cout << "Moving forward by 1/2 rotation" << std::endl;
-    motor->go_for(1000, 0.5);
+        std::vector<vs::ResourceName>* resource_names = robot->resource_names();
 
-    pos = motor->get_position();
-    std::cout << "Motor position " << pos << std::endl;
+        cout << "Resources of the robot:" << endl;
+        for (vs::ResourceName resource : *resource_names) {
+            cout << " - " << resource.name() << " (" << resource.subtype() << ")" << endl;
+        }
+
+        std::string motor_name("motor1");
+
+        cout << "Getting motor: " << motor_name << endl;
+        std::shared_ptr<vs::MotorClient> motor;
+        try {
+            motor = robot->resource_by_name<vs::MotorClient>(motor_name);
+        } catch (const std::exception& e) {
+            cerr << "Failed to find " << motor_name << ". Exiting." << endl;
+            throw;
+        }
+
+        // Motors are always turned off if there is no attached client so this is expected to be 0
+        cout << "Motor power: " << motor->get_power_status().power_pct << endl;
+        cout << "Setting motor power to 50%" << endl;
+        motor->set_power(0.5);
+        cout << "Motor power: " << motor->get_power_status().power_pct << endl;
+
+        print_motor_position(motor);
+
+        cout << "Moving motor by 0.5 rotations at 10rpm" << endl;
+        motor->go_for(10, 0.5);
+        print_motor_position(motor);
+        cout << "Moving motor back to starting location" << endl;
+        if (motor->get_properties().position_reporting) {
+            // More accurate
+            motor->go_to(10, 0);
+        } else {
+            // If position reporting is unavailable, we cannot call go_to
+            // but we can defer to reverting past actions
+            motor->go_for(10, -0.5);
+        }
+        print_motor_position(motor);
+        cout << "Moving motor forward for 2 seconds at 30rpm" << endl;
+        // With `revolutions`==0, the call will be non-blocking
+        // and the motor will rotate indefintely.
+        motor->go_for(30, 0);
+
+        print_motor_position(motor);
+
+        // Wait for 2 seconds, print position every 0.5 seconds
+        for (int i = 0; i < 4; i++) {
+            struct timespec waittime, remaining;
+            waittime.tv_sec = 0;
+            waittime.tv_nsec = 500000000;  // 500ms
+            nanosleep(&waittime, &remaining);
+            print_motor_position(motor);
+        }
+
+        // Explicitly stop the motor
+        cout << "Stopping motor" << endl;
+        vs::AttributeMap stop_map =
+            std::make_shared<std::unordered_map<std::string, std::shared_ptr<vs::ProtoType>>>();
+        motor->stop(stop_map);
```

Comment:
This is unfortunate.

Reply:
We do expose an overload of `stop` in `resource_base.cpp` that takes no argument, so we could move this into `motor` and have the argless version just generate a map, hiding the ugliness from the user.

It's also worth noting that the implementation of `stop(...)` in `motor/client.cpp` doesn't actually use the `AttributeMap` currently. So we could just change that to be an override of the argless `stop()` overload.